src/ot1d.jl

```diff
@@ -0,0 +1,283 @@
+"""
+otCost1d(c,μ::Distributions.UnivariateDistribution,
+           ν::Distributions.UnivariateDistribution)
+
+Calculates the Optimal Transport Cost between μ to ν, where
+they are 1-Dimensional distributions and the cost
+function is of the form ``c(x,y) = h(|x-y|)`` such that
+``h`` is a convex function.
+
+Parameters:\n
+c: The cost function, which should be of the form ``c(x,y) = h(abs(x-y))``
+where `h` is a convex function.
+μ: 1-D distribution (e.g. `μ = Distributions.Normal(0,1)`)\n
+ν: 1-D distribution (e.g. `μ = Distributions.Normal(0,1)`)\n
+"""
+function otCost1d(c,μ::Distributions.UnivariateDistribution,ν::Distributions.UnivariateDistribution)
+    g(μ,ν,x) = c(quantile(μ,x),quantile(ν,x))
+    f(x) = g(μ,ν,x)
+    return quadgk(f,0,1)[1]
+end
+
+"""
+otPlan1d(c,μ::Distributions.UnivariateDistribution,
+            ν::Distributions.UnivariateDistribution)
+
+Calculates the Optimal Transport Plan between μ to ν, where
+they are 1-Dimensional distributions and the cost
+function is of the form ``c(x,y) = h(|x-y|)`` such that
+``h`` is a convex function.
+
+Parameters:\n
+c: The cost function, which should be of the form ``c(x,y) = h(|x-y|)``
+where ``h`` is a convex function.
+μ: 1-D distribution (e.g. `μ = Distributions.Normal(0,1)`)\n
+ν: 1-D distribution (e.g. `μ = Distributions.Normal(0,1)`)\n
+
+Returns the Optimal Transport Plan as a function
+  ```math
+  T(x)=F_\\nu^{-1}(F_\\mu(x)).
+  ```
+"""
+function otPlan1d(c,μ::Distributions.UnivariateDistribution,ν::Distributions.UnivariateDistribution)
+  T(x) = Distributions.quantile(ν,Distributions.cdf(μ,x))
+  return T
+end
+
+
+"""
+otCostPlan1d(c,μ::Distributions.UnivariateDistribution,
+            ν::Distributions.UnivariateDistribution)`
+
+Calculates the Optimal Transport Cost between μ to ν, where
+they are 1-Dimensional distributions and the cost
+function is of the form ``c(x,y) = h(|x-y|)`` such that
+``h`` is a convex function.
+
+Parameters:\n
+c: The cost function, which should be of the form ``c(x,y) = h(abs(x-y))``
+where ``h`` is a convex function.
+μ: 1-D distribution (e.g. `μ = Distributions.Normal(0,1)`)\n
+ν: 1-D distribution (e.g. `μ = Distributions.Normal(0,1)`)\n
+
+Returns the Optimal Transport Cost and the Plan as a function
+  ```math
+  T(x)=F_\\nu^{-1}(F_\\mu(x)).
+  ```
+"""
+function otCostPlan1d(c,μ::Distributions.UnivariateDistribution,ν::Distributions.UnivariateDistribution)
+    g(μ,ν,x) = c(quantile(μ,x),quantile(ν,x))
+    f(x) = g(μ,ν,x)
+    T(x) = Distributions.quantile(ν,Distributions.cdf(μ,x))
+    return quadgk(f,0,1)[1], T
+end
```

Comment:
I don't think this function should be added since it is not more efficient than just computing the cost and map separately.
```suggestion
```
In any case, all suggestions from above apply here as well.